client_plugin/drivers/shared/etcdops.go

```diff
@@ -280,3 +288,122 @@ func addrToEtcdClient(addr string) (*etcdClient.Client, error) {
 
 	return cli, nil
 }
+
+// Update or Create volume metadata in KV store
+func (d *VolumeDriver) writeVolMetadata(name string,
+	record VolumeMetadata) error {
+
+	var msg string
+	var value string
+	var err error
+
+	value = fmt.Sprintf("%d:", record.port)
+	value += record.username + ":"
+	value += record.password + ":"
+	for _, element := range record.clientList {
+		value += element
+		value += "?"
+	}
+
+	etcdAPI := d.createEtcdClient()
+	defer etcdAPI.Close()
+
+	// Write grefc
```

Comment:
I can assume grefc is global ref counter. However, the struct field needs to be commented, and please, do not save 
a few key strokes in the comments. `// update global ref counter'` is probably what you wanted to say here. Also I suggest separating this in a function (optional) to improve readability , writeVolMetadata() is getting too long